src/main/java/org/apache/datasketches/theta/Sketch.java

```diff
@@ -57,125 +53,140 @@
-    if (serVer == 3) {
-      return heapifyFromMemory(srcMem, seed);
-    }
-    if (serVer == 1) {
-      return ForwardCompatibility.heapify1to3(srcMem, seed);
-    }
-    if (serVer == 2) {
-      return ForwardCompatibility.heapify2to3(srcMem, seed);
+    final byte familyID = srcMem.getByte(FAMILY_BYTE);
+    final Family family = idToFamily(familyID);
+    if (family == Family.COMPACT) {
+      return CompactSketch.heapify(srcMem, seed);
     }
-    throw new SketchesArgumentException("Unknown Serialization Version: " + serVer);
+    return heapifyUpdateFromMemory(srcMem, seed);
   }
 
   /**
-   * Wrap takes the sketch image in Memory and refers to it directly. There is no data copying onto
-   * the java heap.  Only "Direct" Serialization Version 3 (i.e, OpenSource) sketches that have
-   * been explicitly stored as direct objects can be wrapped. This method assumes the
-   * {@link org.apache.datasketches.Util#DEFAULT_UPDATE_SEED}.
-   * <a href="{@docRoot}/resources/dictionary.html#defaultUpdateSeed">Default Update Seed</a>.
-   * @param srcMem an image of a Sketch where the image seed hash matches the default seed hash.
-   * <a href="{@docRoot}/resources/dictionary.html#mem">See Memory</a>
+   * Wrap takes the sketch image in the given Memory and refers to it directly.
+   * There is no data copying onto the java heap.
+   * The wrap operation enables fast read-only merging and access to all the public read-only API.
+   *
+   * <p>Only "Direct" Serialization Version 3 (i.e, OpenSource) sketches that have
+   * been explicitly stored as direct sketches can be wrapped.
+   * Wrapping earlier serial version sketches will result in a on-heap CompactSketch
+   * where all data will be copied to the heap. These early versions were never designed to
+   * "wrap".</p>
+   *
+   * <p>Wrapping any subclass of this class that is empty or contains only a single item will
+   * result in on-heap equivalent forms of empty and single item sketch respectively.
+   * This is actually faster and consumes less overall memory.</p>
+   *
+   * <p>For Update Sketches this method checks if the
+   * <a href="{@docRoot}/resources/dictionary.html#defaultUpdateSeed">Default Update Seed</a></p>
+   * was used to create the source Memory image.
+   *
+   * <p>For Compact Sketches this method assumes that the sketch image was created with the
+   * correct hash seed, so it is not checked.</p>
+   *
+   * @param srcMem an image of a Sketch.
+   * <a href="{@docRoot}/resources/dictionary.html#mem">See Memory</a>.
    * @return a Sketch backed by the given Memory
    */
   public static Sketch wrap(final Memory srcMem) {
-    return wrap(srcMem, DEFAULT_UPDATE_SEED);
+    final int  preLongs = srcMem.getByte(PREAMBLE_LONGS_BYTE) & 0X3F;
+    final int serVer = srcMem.getByte(SER_VER_BYTE) & 0XFF;
+    final int familyID = srcMem.getByte(FAMILY_BYTE) & 0XFF;
+    final Family family = Family.idToFamily(familyID);
+    if (family == Family.QUICKSELECT) {
```

Comment:
Is it possible to have family == ALPHA here?

Reply:
No. The Alpha sketch was never intended to be wrapped.